rules/CVE-2021-25742/raw.rego

```diff
@@ -29,78 +37,32 @@ is_nginx_image(image) {
 	contains(image, "ingress-nginx")
 }
 
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    startswith(tag, "v")
-    tag <= "v0.49"
-}
-	
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    startswith(tag, "v")
-    tag  == "v1.0.0"
-}
-
-is_vulnerable(image, namespace) {
-	not contains(image, "@")
-	version := split(image, ":")
-	tag := version[count(version)-1]
-    startswith(tag, "v")
-	tag <= "v0.49"
+is_allow_snippet_annotation_on(namespace) {
+    configmaps := [configmap | configmap = input[_]; configmap.kind == "ConfigMap"]
+	configmap_on_ingress_namespace := [configmap |  configmap= configmaps[_]; configmap.metadata.namespace == namespace]
+	config_maps_with_snippet := [configmap |  configmap= configmap_on_ingress_namespace[_];  configmap.data["allow-snippet-annotations"] == "false"]
+	count(config_maps_with_snippet) < 1
 }
 
-is_vulnerable(image, namespace) {
-	not contains(image, "@")
-	version := split(image, ":")
-	tag := version[count(version)-1]
-    startswith(tag, "v")
-	tag  == "v1.0.0"
+is_vulnerable(image_version, namespace) {
+	image_version[0] == 0
+	image_version[1] < 49
+	is_allow_snippet_annotation_on(namespace)
 }
 
-###### without 'v'
-	
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    not startswith(tag, "v")
-    tag <= "0.49"
+is_vulnerable(image_version, namespace) {
+	image_version[0] == 0
+	image_version[1] == 49
+	image_version[2] == 0
+	is_allow_snippet_annotation_on(namespace)
 }
 	
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    not startswith(tag, "v")
-    tag  == "1.0.0"
-}
-
-is_vulnerable(image, namespace) {
-	not contains(image, "@")
-	version := split(image, ":")
-	tag := version[count(version)-1]
-    not startswith(tag, "v")
-	tag <= "0.49"
+is_vulnerable(image_version, namespace) {
+	image_version[0] == 1
+	image_version[1] == 0
+	image_version[2] == 0
+	is_allow_snippet_annotation_on(namespace)
 }
-is_vulnerable(image, namespace) {
-	not contains(image, "@")
-	version := split(image, ":")
-	tag := version[count(version)-1]
-    not startswith(tag, "v")
-	tag  == "1.0.0"
-}
-
-is_vulnerable(image, namespace) {
-    configmaps := [configmap | configmap = input[_]; configmap.kind == "ConfigMap"]
-	configmap_on_ingress_namespace := [configmap |  configmap= configmaps[_]; configmap.metadata.namespace == namespace]
-	config_maps_with_snippet := [configmap |  configmap= configmap_on_ingress_namespace[_];  configmap.data["allow-snippet-annotations"] == "false"]
-	count(config_maps_with_snippet) < 1
-}
-
 
 is_tag_image(image) {
```

Comment:
Without real image name example it is difficult to say if this is correct and what it tries to check.